fs/operations/dedupe.go

```diff
@@ -18,16 +18,23 @@ import (
 )
 
 // dedupeRename renames the objs slice to different names
-func dedupeRename(remote string, objs []fs.Object) {
-	f := objs[0].Fs()
-	doMove := f.Features().Move
+func dedupeRename(f fs.Fs, remote string, objs []fs.Object) {
+	info := objs[0].Fs()
+	doMove := info.Features().Move
 	if doMove == nil {
-		log.Fatalf("Fs %v doesn't support Move", f)
+		log.Fatalf("Fs %v doesn't support Move", info)
 	}
 	ext := path.Ext(remote)
 	base := remote[:len(remote)-len(ext)]
 	for i, o := range objs {
-		newName := fmt.Sprintf("%s-%d%s", base, i+1, ext)
+		suffix := 1
+		newName := fmt.Sprintf("%s-%d%s", base, i+suffix, ext)
+		_, err := f.NewObject(newName)
+		for err != fs.ErrorObjectNotFound {
```

Comment:
Being super paranoid I would put a limit on this loop - say 10 or 100 times maximum!